pkg/reconciler/autoscaling/kpa/kpa.go

```diff
@@ -260,40 +260,35 @@ func reportMetrics(pa *autoscalingv1alpha1.PodAutoscaler, pc podCounts) {
 //	| -1   | >= min | >0    | active     | active     |
 func computeActiveCondition(ctx context.Context, pa *autoscalingv1alpha1.PodAutoscaler, pc podCounts) {
 	minReady := activeThreshold(ctx, pa)
+
 	if pc.ready >= minReady && pa.Status.ServiceName != "" {
 		pa.Status.MarkScaleTargetInitialized()
 	}
 
 	switch {
-	// Need to check for minReady = 0 because in the initialScale 0 case, pc.want will be -1.
-	case pc.want == 0 || minReady == 0:
-		if pa.Status.IsActivating() && minReady > 0 {
-			// We only ever scale to zero while activating if we fail to activate within the progress deadline.
-			pa.Status.MarkInactive("TimedOut", "The target could not be activated.")
-		} else {
-			pa.Status.MarkInactive(noTrafficReason, "The target is not receiving traffic.")
-		}
+	case pa.Status.IsActivating() && pc.want == 0 && minReady > 0:
+		pa.Status.MarkInactive("TimedOut", "The target could not be activated.")
 
-	case pc.ready < minReady:
-		if pc.want > 0 || !pa.Status.IsInactive() {
-			pa.Status.MarkActivating(
-				"Queued", "Requests to the target are being buffered as resources are provisioned.")
-		} else {
-			// This is for the initialScale 0 case. In the first iteration, minReady is 0,
-			// but for the following iterations, minReady is 1. pc.want will continue being
-			// -1 until we start receiving metrics, so we will end up here.
-			// Even though PA is already been marked as inactive in the first iteration, we
-			// still need to set it again. Otherwise reconciliation will fail with NewObservedGenFailure
-			// because we cannot go through one iteration of reconciliation without setting
-			// some status.
-			pa.Status.MarkInactive(noTrafficReason, "The target is not receiving traffic.")
-		}
+	case !pa.Status.IsInactive() && (pc.want == 0 || minReady == 0):
```

Comment:
I'm skeptical of making changes here cause because there is a lot of nuance and I'm guessing there's some gap in our testing so we won't catch all the issues

eg. based on the prior logic this should really just be `pc.want == 0 || minReady == 0` - there's no active condition check

Reply:
I think to reason about this we should take the above table and highlight the changes we are making to it since you're trying to introduce reachability into this decision